R/jira.R

```diff
@@ -8,103 +8,392 @@
-  ext_info_cell[["resolution"]][["name"]] <- list("Fixed")
-
-  ext_info_cell[["resolutiondate"]] <- list("2007-08-13T19:12:33.000+0000")
-
-  ext_info_cell[["assignee"]] <- list()
-
-  ext_info_cell[["updated"]] <- list("2008-05-12T08:01:39.000+0000")
-
-  ext_info_cell[["status"]] <- list()
-  ext_info_cell[["status"]][["self"]] <- list("https://domain.org/jira/rest/api/2/status/6")
-  ext_info_cell[["status"]][["description"]] <- list("The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.")
-  ext_info_cell[["status"]][["iconUrl"]] <- list("https://domain.org/jira/images/icons/statuses/closed.png")
-  ext_info_cell[["status"]][["name"]] <- "Closed"
-  ext_info_cell[["status"]][["id"]] <- "6"
-  ext_info_cell[["status"]][["statusCategory"]] <- list()
-  ext_info_cell[["status"]][["statusCategory"]][["self"]] <- list("https://domain.org/jira/rest/api/2/statuscategory/3")
-  ext_info_cell[["status"]][["statusCategory"]][["id"]] <- list(3)
-  ext_info_cell[["status"]][["statusCategory"]][["key"]] <- list("done")
-  ext_info_cell[["status"]][["statusCategory"]][["colorName"]] <- list("green")
-  ext_info_cell[["status"]][["statusCategory"]][["name"]] <- list("Done")
-
+  ext_info_cell <- create_ext_info_issues()
   issues[["ext_info"]][[1]] <- ext_info_cell
 
   jira_json_path <- file.path(folder_path,"fake_issues.json")
   jsonlite::write_json(issues,file.path(folder_path,"fake_issues.json"))
 
+  return(jira_json_path)
+}
+
+#' Create base_info_cell
+#'
+#' Creates and formats base_info_cell for fake Jira issue
+#'
+#' @param id ID of Jira issue
+#' @param self URL of API endpoint
+#' @param key key for Jira issue
+#' @param JirAgileR_id JirAgileR id
+#' @return A list named 'base_info_cell' containing all the parameters
+#' @export
+create_base_info <- function(id = 11, self = "https://project.org/jira/rest/api/latest/issue/11", key = "PROJECT-68", JirAgileR_id = 1) {
+  base_info_cell <- list(
+    id = id,
+    self = self,
+    key = key,
+    JirAgileR_id = JirAgileR_id
+  )
+
+  return(base_info_cell)
+}
+
+#' Create ext_info_cell for issues (without comments)
+#'
+#' Creates and formats ext_info_cell for fake Jira issue
+#'
+#' @return ext info cell
+#' @export
+create_ext_info_issues <- function() {
+  ext_info_cell <- list(
+    summary = list("Issue Description"),
+    issuetype = create_issuetype(),
+    components = create_components(component_details),
+    creator = create_creator(),
+    created = list("2007-07-08T06:07:06.000+0000"),
+    description = list(),
+    reporter = create_reporter(),
+    resolution = create_resolution(self_url = "https://domain.org/jira/rest/api/2/resolution/1"),
+    resolutiondate = list("2007-08-13T19:12:33.000+0000"),
+    assignee = list(),
+    updated = list("2008-05-12T08:01:39.000+0000"),
+    status = create_status()
+  )
+
+  return(ext_info_cell)
+}
+
+#' Create Issue Type
+#'
+#' Create issue type cell for fake Jira issue
```

Comment:
Too short description again. You have to consider what this information is for, and where it is from, and guide your user here. This will require some research on your part, or experimenting with Kaiaulu issue tracker. As an example, you can see this block specify the issue type the user chose. You should, therefore, tell in this description an issue type can be a bug, an enhancement, etc and also specify `\url{https://support.atlassian.com/jira-cloud-administration/docs/what-are-issue-types/}` and `\url{https://community.atlassian.com/t5/Jira-articles/Understanding-issue-types-in-jira/ba-p/1497237}` as relevant documentation.

Basically, your function should give a bit of context to the user on how they can type other inputs. You do not need to explain all possibilities, just one (your e.g.) and reference them where they can learn more to write others on their own. Your description does need to guide the user on making sure it is consistent and ideally, enforce that via your code. For example, above, don't ask the user the id twice, where they can make an inconsistent parameter. But rather, ask once, and reinforce on your code logic they are the same information.